From 27a8a34f4896e70e0140fb72584cb8df448c72db Mon Sep 17 00:00:00 2001
From: Markus Armbruster <armbru@redhat.com>
Date: Thu, 17 Feb 2011 09:32:42 -0200
Subject: [PATCH 06/13] blockdev: New drive_get_by_blockdev() (v2)

RH-Author: Markus Armbruster <armbru@redhat.com>
Message-id: <1297935168-2576-7-git-send-email-armbru@redhat.com>
Patchwork-id: 18384
O-Subject: [PATCH RHEL6.0.Z qemu-kvm 06/12] blockdev: New
	drive_get_by_blockdev() (v2)
Bugzilla: 677170
RH-Acked-by: Marcelo Tosatti <mtosatti@redhat.com>
RH-Acked-by: Kevin Wolf <kwolf@redhat.com>
RH-Acked-by: Jes Sorensen <Jes.Sorensen@redhat.com>

From: Markus Armbruster <armbru@redhat.com>

BZ: 654682
Upstream-status: accepted

Signed-off-by: Markus Armbruster <armbru@redhat.com>
Signed-off-by: Kevin Wolf <kwolf@redhat.com>
(cherry picked from commit e4700e595ea0b24d5291dbd68deba26d7a955703)
Signed-off-by: Anthony Liguori <aliguori@us.ibm.com>
--
v1 -> v2
 - Reorder declarations to match upstream

Signed-off-by: Eduardo Habkost <ehabkost@redhat.com>
---
 sysemu.h |    1 +
 vl.c     |   12 ++++++++++++
 2 files changed, 13 insertions(+), 0 deletions(-)

Signed-off-by: Luiz Capitulino <lcapitulino@redhat.com>
---
 sysemu.h |    1 +
 vl.c     |   12 ++++++++++++
 2 files changed, 13 insertions(+), 0 deletions(-)

diff --git a/sysemu.h b/sysemu.h
index 8c3ec8e..7faa8e9 100644
--- a/sysemu.h
+++ b/sysemu.h
@@ -194,6 +194,7 @@ extern DriveInfo *drive_get(BlockInterfaceType type, int bus, int unit);
 extern DriveInfo *drive_get_by_id(const char *id);
 extern int drive_get_max_bus(BlockInterfaceType type);
 extern void drive_uninit(DriveInfo *dinfo);
+extern DriveInfo *drive_get_by_blockdev(BlockDriverState *bs);
 extern const char *drive_get_serial(BlockDriverState *bdrv);
 
 extern BlockInterfaceErrorAction drive_get_on_error(
diff --git a/vl.c b/vl.c
index 5014a5a..8b9dbc5 100644
--- a/vl.c
+++ b/vl.c
@@ -2078,6 +2078,18 @@ DriveInfo *drive_get_by_id(const char *id)
     return NULL;
 }
 
+DriveInfo *drive_get_by_blockdev(BlockDriverState *bs)
+{
+    DriveInfo *dinfo;
+
+    QTAILQ_FOREACH(dinfo, &drives, next) {
+        if (dinfo->bdrv == bs) {
+            return dinfo;
+        }
+    }
+    return NULL;
+}
+
 int drive_get_max_bus(BlockInterfaceType type)
 {
     int max_bus;
-- 
1.7.4.rc1.16.gd2f15e

